app/models/daily.rb

```diff
@@ -6,4 +6,22 @@ class Daily < ApplicationRecord
   has_many :scores, dependent: :destroy
   has_one :supporter, dependent: :destroy
 
+
+  def weeks_scores
+    results = self.scores.created_between((Date.today - 7), Date.today).order(date: :desc).pluck(:score)
+
+    while results.length < 7
+      results.unshift(nil)
+    end
+    results
+  end
+
+  def weeks_average
+    results = self.scores.created_between((Date.today - 7), Date.today).order(date: :desc).pluck(:score)
+    results.reduce(:+)/results.length unless results.empty?
+  end
+
+  def table_row
+    [[goal] + weeks_scores.map {|n| n || 0}]
+  end
 end
```

Comment:
@glassjoseph if it's purely for formatting, then it makes more sense in a helper method. your helper methods are intended to help you keep your views clean. this is a perfect example of where you'd most likely use a helper method rather than putting it in the model. the model is typically for extracting logic related to the data 👍